src/db.js

```diff
@@ -168,6 +205,16 @@ class Db {
 		return cacheObj;
 	}
 
+	async _cacheAllWithOpts(m, q, f, o) {
+		let cacheKey = this.getCacheKeyWithOpts(m, q, f, o) + "_all";	// add suffix so as to distinguise between cache first and cacheall key if rest of parameters are the same
+		let cacheObj = await this.cache.get(cacheKey);
+		if (cacheObj === undefined) {
+			cacheObj = await this.findAll(m, q, f, o);
+			await this.cache.set(cacheKey, cacheObj, 10000);
```

Comment:
since you are taking opts as an argument then let's also take the cache time in the opts

```suggestion
			await this.cache.set(cacheKey, cacheObj, o.timeout || 10000);
```